native/src/widget/tooltip.rs

```diff
@@ -0,0 +1,239 @@
+//! Display a widget over another.
+use std::hash::Hash;
+
+use crate::{
+    layout, overlay, Clipboard, Element, Event, Hasher, Layout, Length, Point,
+    Size, Vector, Widget,
+};
+
+/// An element to display a widget over another.
+#[allow(missing_debug_implementations)]
+pub struct Tooltip<'a, Message, Renderer: self::Renderer> {
+    state: State,
+    content: Element<'a, Message, Renderer>,
+    hover_content: Option<Element<'a, Message, Renderer>>,
+}
+
+impl<'a, Message, Renderer> Tooltip<'a, Message, Renderer>
+where
+    Renderer: self::Renderer,
+{
+    /// Creates an empty [`Tooltip`].
+    ///
+    /// [`Tooltip`]: struct.Tooltip.html
+    pub fn new<T>(content: T) -> Self
+    where
+        T: Into<Element<'a, Message, Renderer>>,
+    {
+        Tooltip {
+            state: Default::default(),
+            content: content.into(),
+            hover_content: None,
+        }
+    }
+
+    /// Sets the hover_content of the [`Tooltip`].
+    ///
+    /// [`Tooltip`]: struct.Tooltip.html
+    pub fn hover_content<T>(mut self, hover_content: T) -> Self
+    where
+        T: Into<Element<'a, Message, Renderer>>,
+    {
+        self.hover_content = Some(hover_content.into());
+        self
+    }
```

Comment:
I think we should make empty tooltips impossible. In other words, we should ask for both the `target` and the `content` of the `Tooltip` in `new`.

Reply:
19fb8caee81bf8856f11772ed47196cbda7b2e3d solves that.